src/wal/src/local_storage_impl/segment.rs

```diff
@@ -454,6 +515,76 @@ impl SegmentManager {
         }
         Ok(())
     }
+
+    pub fn mark_delete_entries_up_to(
+        &self,
+        location: WalLocation,
+        sequence_num: SequenceNumber,
+    ) -> Result<()> {
+        let current_segment_id = self.current_segment.lock().unwrap().lock().unwrap().id;
+        let mut segments_to_remove: Vec<u64> = Vec::new();
+        let mut all_segments = self.all_segments.lock().unwrap();
+
+        for (_, segment) in all_segments.iter() {
+            let mut segment = segment.lock().unwrap();
+
+            segment.mark_deleted(location.table_id, sequence_num);
```

Comment:
Since the seq in once segment is increasing among all tables, so we can just compare the `sequence_num` with its min/max segment of one segment?